packages/firebase_performance/test/firebase_performance_test.dart

```diff
@@ -0,0 +1,211 @@
+
+        testTrace.incrementCounter('counter3', 25);
+        expect(testTrace.counters, <String, int>{
+          'counter1': -1,
+          'counter2': 1,
+          'counter3': 25,
+        });
+
+        await testTrace.start();
+        await testTrace.stop();
+        testTrace.incrementCounter('counter3');
+        expect(testTrace.counters, <String, int>{
+          'counter1': -1,
+          'counter2': 1,
+          'counter3': 25,
+        });
+      });
+
+      test('newTrace', () async {
+        await performance.newTrace('test-trace');
+        expect(invokedMethod, 'FirebasePerformance#newTrace');
+        expect(arguments, 'test-trace');
+      });
+
+      test('startTrace', () async {
+        await performance.startTrace('test-trace');
+        expect(invokedMethod, 'Trace#start');
+        expect(arguments, null);
+      });
+
+      test('start', () async {
+        await testTrace.start();
+        expect(invokedMethod, 'Trace#start');
+        expect(arguments, null);
+        expect(testTrace.hasStarted, true);
+      });
+
+      test('stop', () async {
+        testTrace.incrementCounter('counter1');
+        testTrace.putAttribute('attr1', 'apple');
+        await testTrace.start();
+        await testTrace.stop();
+
+        expect(invokedMethod, 'Trace#stop');
+        expect(arguments, <String, dynamic>{
+          'id': null,
+          'name': 'test',
+          'counters': <String, int>{'counter1': 1},
+          'attributes': <String, String>{'attr1': 'apple'},
+        });
+        expect(testTrace.hasStarted, true);
+        expect(testTrace.hasStopped, true);
+      });
+
+      test('start and stop called in wrong order', () async {
+        Trace trace = await performance.newTrace("test");
+        await trace.stop();
+        expect(trace.hasStarted, false);
+        expect(trace.hasStopped, false);
+
+        trace = await performance.newTrace("test");
+        await trace.start();
+        await trace.start();
+        expect(trace.hasStarted, true);
+        expect(trace.hasStopped, false);
+
+        trace = await performance.newTrace("test");
+        await trace.start();
+        await trace.stop();
+        await trace.stop();
+        expect(trace.hasStarted, true);
+        expect(trace.hasStopped, true);
+      });
+
+      test('putAttribute', () async {
+        testTrace.putAttribute('attr1', 'apple');
+        testTrace.putAttribute('attr2', 'are');
+        expect(testTrace.attributes, <String, String>{
+          'attr1': 'apple',
+          'attr2': 'are',
+        });
+
+        testTrace.putAttribute('attr1', 'delicious');
+        expect(testTrace.attributes, <String, String>{
+          'attr1': 'delicious',
+          'attr2': 'are',
+        });
+
+        await testTrace.start();
+        await testTrace.stop();
+        testTrace.putAttribute('attr3', 'yes');
+        expect(testTrace.attributes, <String, String>{
+          'attr1': 'delicious',
+          'attr2': 'are',
```

Comment:
I wasn't able to figure out why this test is passing, shouldn't `'attr3'` be in there?